src/librbd/managed_lock/GetLockerRequest.cc

```diff
@@ -103,14 +103,14 @@ void GetLockerRequest<I>::handle_get_lockers(int r) {
     return;
   }
 
-  m_locker->entity = iter->first.locker;
-  m_locker->cookie = iter->first.cookie;
-  m_locker->address = iter->second.addr.get_legacy_str();
-  if (m_locker->cookie.empty() || m_locker->address.empty()) {
-    ldout(m_cct, 20) << "no valid lockers detected" << dendl;
+  if (iter->second.addr.is_blank_ip()) {
+    ldout(m_cct, 5) << "locker has a blank address" << dendl;
     finish(-ENOENT);
```

Comment:
I think you missed this part:
> Treating a locker that has a valid cookie but a blank address as if there were no lockers seems wrong.
```suggestion
    finish(-EBUSY);
```

Reply:
Why does a blank address suggest that the locker is busy or used by other mechanism? Isn't `ENOENT` a better fit?

Reply:
This method finishes `GetLockerRequest` with `ENOENT` if there are no lockers (see "no lockers detected" above).  If there is a locker with a valid formatted cookie but a blank address, something fishy must be going on.  Equating that to the genuine "no lockers" case is unlikely to lead to anything good.

Reply:
Ack, updated.